cocotb/scheduler.py

```diff
@@ -725,18 +699,7 @@ def _trigger_from_any(self, result) -> Trigger:
         if isinstance(result, cocotb.triggers.Waitable):
             return self._trigger_from_waitable(result)
 
-        if sys.version_info >= (3, 6) and inspect.isasyncgen(result):
-            raise TypeError(
-                "{} is an async generator, not a coroutine. "
-                "You likely used the yield keyword instead of await.".format(
-                    result.__qualname__))
-
-        raise TypeError(
-            "Coroutine yielded an object of type {}, which the scheduler can't "
-            "handle: {!r}\n"
-            "Did you forget to decorate with @cocotb.coroutine?"
-            .format(type(result), result)
-        )
+        return self._trigger_from_unstarted_coro(cocotb.decorators.make_task(result))
```

Comment:
I am, but you're right, it's a bad idea here because it will construct naked generators into tasks, same with `scheduler.add`. Perhaps this whole PR is junk...